### PR TITLE
Archive Task page functionality

### PR DESCRIPTION
**Pull request by Rolo123y**
**Archived page:**
- Delete Task: Users can now delete tasks that are archived. To delete a task, it MUST be archived first!
- UnArchive Task: Users can now un-archive tasks that are in the archive page.

**ListView page:**
- Archive Task: Users are able to archive tasks of their choosing. The task disappears from the list and gets placed into the Archived page.
- Removed iframe for task viewing and changed it to modals. This change was brought from a group discussion as iframes are confusing and hard to work with while modals are simpler and are used throughout the application.

**SelectBoard page:**
- Re-aligned the 'Create Board' button so it matches with the boards.

**Things to test:**
- Archiving a task and check for it in the Archive page.
- UnArchive the same task and check for it in the ListView page.
- Delete a task in the Archive page and see that the task is deleted in the database.
- General flow of Archiving/unArchiving/Deleting.

### CODE REVIEW

**Comment by AaronDzf** [comment]
Overall the archive board is looking good. A few initial notes:

Tested the method for:
- Archiving a task from modal will properly put it in archive board. Archive board is universal across boards: not sure if this was intentional, let me know.
- Unarchived task goes back to the appropriate task list and board where it was archived from 
- Delete task from archive board is functioning

Suggestions:
- In future, grey out textboxes/user input fields for archived tasks in archive board so users aren't confused about whether the values can be edited or not.

**Review by AaronDzf — DISMISSED** [review]
Good changes overall, functions on my my machine

File: ArchivedPage.php [hunk 1]
```diff
@@ -87,24 +127,122 @@
...
+              <?php if ($edit) { ?><td><input type="text" class="form-control" name="typeOfWorkUpdate" value="<?php echo $task['typeOfWork']; ?>"></td><?php } ?>
+            </tr>
+            <tr>
+              <th scope="row">Course</th>
+              <td><?php echo $task['courseTitle'];?></td>
+              <?php if ($edit) { ?><td><select name = "updateCourseID">
+                    <option value="<?php echo $task['courseID']; ?>"><?php echo $task['courseTitle']; ?></option>
+                        <?php                    
+                            $query = "SELECT courseID,courseTitle FROM course";
+                            $courseTitles = mysqli_query($conn, $query);
```
Comment (gaurav613) [inline ArchivedPage.php:203]: To maintain consistency, please use PDO to execute query calls - this file has already made use of it(lines 134 to 147).

File: ArchivedPage.php [hunk 2]
```diff
@@ -87,24 +127,122 @@
...
+              </select></td><?php } ?>
+            </tr>
+            <tr>
+              <th scope="row">List</th>
+              <td><?php echo $task['listTitle'];?></td>
+              <?php if ($edit) { ?><td><select name = "updateListID">
+                    <option value="<?php echo $task['listID']; ?>"><?php echo $task['listTitle']; ?></option>
+                        <?php                    
+                            $query = "SELECT listID,listTitle FROM taskList";
+                            $taskLists = mysqli_query($conn, $query);
```
Comment (gaurav613) [inline ArchivedPage.php:217]: Same here, in regards to using PDO for maintaining consistent code.

File: listView.php [hunk 3]
```diff
@@ -484,20 +514,152 @@
...
+              </select></td><?php } ?>
+            </tr>
+            <tr>
+              <th scope="row">List</th>
+              <td><?php echo $task['listTitle'];?></td>
+              <?php if ($edit) { ?><td><select name = "updateListID">
+                    <option value="<?php echo $task['listID']; ?>"><?php echo $task['listTitle']; ?></option>
+                        <?php                    
+                            $query = "SELECT listID,listTitle FROM taskList";
+                            $taskLists = mysqli_query($conn, $query);
```
Comment (gaurav613) [inline listView.php:623]: Can also be changed to PDO - along with lines 610-614.

**Review by gaurav613 — CHANGES_REQUESTED** [review]
This looks really nice! Tasks are getting archived properly, and the archived list is getting displayed well. Even unarchiving tasks returns them to the same list.

Some feedback regarding improvement:

- In the Archived List page, it looks like there are 2 "default" tasks(not archived by me) that are getting shown, and I'm not able to view/delete them, though the mouse changes to a pointer when hovering over them. 
- It's allowing me to change the properties of a task in the archive page, but not allowing me to update them - this can be solved either by adding an update button, or limiting the user's input from this page.
- The code's consistency can be improved, with respect to query calling(using PDO instead of mysqli) - I have pointed out individual instances below.

Overall, this is some solid functionality, which will definitely be helpful to users! Good work! 👍

**Comment by gaurav613** [comment]
I am merging this for now - the issue of new tasks being shown requires a simple query change(it's pulling all tasks which are flagged as archived, instead of being conditioned on the board ID), which can be performed later.
The code consistency can also be improved in the next sprint.
Other than that, everything looks good 👍

**Review by gaurav613 — APPROVED** [review]

**Pull request merged**